src/main/java/com/eprosima/idl/parser/tree/TreeNode.java

```diff
@@ -118,6 +118,31 @@ public String getNamespace()
         return namespace;
     }
 
+    /**
+     * @ingroup api_for_stg
+     * @brief This function returns the name to use in the type object registration methods.
+     * @return A string with the name to use in the type object registration methods.
+     */
+    public String getNameForTypeObjectRegistration()
+    {
+        return interface_prefix() + getName();
+    }
+
+    protected String interface_prefix()
```

Comment:
NIT: Perhaps we should use camel case to unify the format.

```suggestion
    protected String interfacePrefix()
```

Reply:
done